src/trace/implementations/rhh.rs

```diff
@@ -742,7 +758,7 @@ mod val_batch {
         fn push(&mut self, ((key, val), time, diff): Self::Item) {
 
             // Perhaps this is a continuation of an already received key.
-            if self.result.keys.last() == Some(&key) {
+            if self.result.keys.last() == Some(key.borrow()) {
```

Comment:
Here is one of the first examples of actually having to relate the types. I'm not even sure we use any methods of `ToOwned` anywhere else!